assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -317,16 +321,37 @@ public String archive(String siteId, Document doc, Stack<Element> stack, String
 
         int assignmentsArchived = 0;
         for (Assignment assignment : getAssignmentsForContext(siteId)) {
+
+            Map<String, Object> content = null;
+            Map<String, Object> tool = null;
+            if ( assignment.getContentId() != null ) {
+                Long contentKey = assignment.getContentId().longValue();
+                content = ltiService.getContent(contentKey.longValue(), siteId);
+                if ( content != null ) {
+                    Long toolKey = Long.valueOf(content.get(LTIService.LTI_TOOL_ID).toString());
+                    if (toolKey != null) {
+                        tool = ltiService.getTool(toolKey, siteId);
+                    }
+                }
+            }
```

Comment:
this adds logic to assignments about what is in the return value of this `Map<String, Object> content`
does it make sense that every non LTI external code, needs to understand what data are in these Maps?

I have this distinct sensation of a service leaking it's internals!!!

Reply:
@ern Someday we will rewrite the LTIService to use Objects.  For now it returns Maps with pre-defined key string names.  This is the way until we refactor the service.

Reply:
isn't the point that this looks like a bunch of LTI-specific code that should live in LTI and not in assignment?

Reply:
@ern cc: @ottenhoff It took me a bit to think about this but I think I misunderstood your comment.  I took them to mean you are leaking data" and defended that.  What I am really leaking is "logic" and "convention" that belongs within the LTI orbit - not within the Assignments orbit.  And frankly you are right because much of this code to make LTI tool imports work was copied and adapted from when Lessons does the same thing when importing an LTI link from an IMS CC.  So by copying the code from Lessons and tweaking it a bit, I have violated the D.R.Y. principle.  I will work on a refactor that moves as much logic as I can from *both* Assignments and Lessons into the LTI world and reduce the LTI related logic in Assignments and Lessons to a minimum.   Thanks.

Reply:
@ern cc: @ottenhoff I refactored the code a lot, reducing the changes in AssignmentServiceImpl to just a couple of lines.  I pulled all that logic into the service.  It makes a lot more sense.  I rebased and forced pushed so I will close this one and you can take another crack it.  Thanks.